Assets/HoloToolkit/Input/README.md

```diff
@@ -3,50 +3,649 @@
+    - [Getting Focus](#getting-focus)
+    - [Getting Clicks and Generic Input](#getting-clicks-and-generic-input)
+    - [Xbox Controller Input](#xbox-controller-input)
+    - [Voice Input](#voice-input)
+    - [Dictation Input](#dictation-input)
+    - [Hold, Navigation, and Manipulation Inputs](#hold-navigation-and-manipulation-inputs)
+    - [Motion Controller Input](#motion-controller-input)
 - [Prefabs](#prefabs)
 - [Scripts](#scripts)
 - [Test Prefabs](#test-prefabs)
 - [Test Scripts](#test-scripts)
 - [Tests](#tests)
 
 ## Overview
+This contains a fully-featured input system, which allows you to handle various types of input and send them to any `GameObject` being currently gazed at, or any fallback object. It also includes a few example cursors that fully leverages the Unity's animation system.
+- This input system uses and extends Unity's default `EventSystem` so there's no need for a custom input module.
+- This input system is cross platform and can be easily extended with custom input sources, events, and input handlers.
+- This input system works with Unity’s native uGUI.
 
-This contains a fully-featured **input system**, which allows you to handle various types of input and send them to any game object being currently gazed at, or any fallback object. It also includes a few example cursors, similar to the HoloLens shell cursor, that fully leverages the Unity's animation system.  This input system uses Unity's default **EventSystem** and there's no need for a custom **input module**.  The default **Standalone Input Module** works with this input system.
+### Input Module Design
 
-## Input System Diagrams:
-![alt text](/External/ReadMeImages/InputSystemDiagram.png)
-![alt text](/External/ReadMeImages/CursorSystemDiagram.PNG)
+Each input source (Motion Controllers, Hands, Gestures, Mouse, Keyboard, etc.) implement a `IInputSource` interface. The interface defines various events that the input sources can trigger. The input sources register themselves with the `InputManager`, whose role it is to forward input to the appropriate `GameObject`s. Input sources can be dynamically enabled / disabled as necessary, and new input sources can be created to support different input devices.
 
-### Input Module Design
-The input module is designed to be extensible: it could support various input mechanisms and various types of gazers.
+The `InputManager` listens to the various events coming from the input sources, and also takes into account which `GameObject` the current gaze and pointers are focused on.
+
+By default, input events are sent to the currently focused `GameObject`, if that object implements the appropriate interface.  Any event sent by the `InputManager` always bubbles up from the focused `GameObject`, to each of its ancestors.
+
+If you wish to be able to send input events to `GameObject`s that do not require gaze (such as voice commands), you will need to register the handler as a Global Listener.
```

Comment:
Hum, I'd say you'd be better off registering as a fallback handler than a global listener.

Global listener is a concept that was added mostly for the sake of the cursor and similar components that always need to receive input events, even if other things handled them. Voice commands would more likely behave properly if implements as fallback, so that if something you are gazing at needs to handle a voice command (think of "Select" for example), it receives the input event properly